projects/netgrif-components/src/lib/data-fields/enumeration-field/enumeration-autocomplete-select-field/enumeration-autocomplete-select-field.component.html

```diff
@@ -3,19 +3,20 @@
     <input type="text"
            aria-label="Autocomplete"
            matInput
-           #input
+           #input="ngModel"
            [placeholder]="enumerationField.placeholder"
-           [formControl]="formControlRef"
            [matAutocomplete]="auto"
+           [(ngModel)]="tmpValue"
            [required]="enumerationField.behavior.required"
+           [disabled]="formControlRef.disabled"
            (focus)="change()"
            (keyup)="change()">
-    <mat-autocomplete [displayWith]="renderSelection" autoActiveFirstOption #auto="matAutocomplete">
+    <mat-autocomplete [displayWith]="renderSelection" autoActiveFirstOption #auto="matAutocomplete" (optionSelected)="select($event)">
         <mat-option [value]="null">---</mat-option>
         <mat-option *ngFor="let option of filteredOptions | async" [value]="option.key" (click)="change()">
             {{option.value}}
         </mat-option>
     </mat-autocomplete>
     <mat-hint>{{enumerationField.description}}</mat-hint>
-    <mat-error *ngIf="enumerationField.isInvalid(formControlRef)">{{buildErrorMessage()}}</mat-error>
+    <mat-error *ngIf="buildErrorMessage() !== undefined">{{buildErrorMessage()}}</mat-error>
```

Comment:
why was this changed?

Reply:
because formControlRef isnt attached to input, so it is impossible to touch that formcontrol, so call "enumerationField.isInvalid(formControlRef)" is always false

Reply:
i have taken a look at it and I believe it would be better to extend/adapt the isInvalid method to ignore touch state if it is a problem.

The `buildErrorMessage` method is not inherited and some components even lack support for certain validations.

The form control is currently the single point of truth regarding field validity and I believe this should be preserved.